### PR TITLE
Make node/edge highlighter objects immutable in app store

### PR DESCRIPTION
**Pull request by davkal**
* refactor
* fixed reverse edge highlighting on node hover

### CODE REVIEW

File: client/app/scripts/stores/app-store.js [hunk 1]
```diff
@@ -433,22 +412,53 @@ export class AppStore extends Store {
...
-        mouseOverNodeId = payload.nodeId;
+        const nodeId = payload.nodeId;
+
+        // clear old highlights
+        highlightedNodeIds = highlightedNodeIds.clear();
+        highlightedEdgeIds = highlightedEdgeIds.clear();
+
+        // highlight nodes
+        highlightedNodeIds = highlightedNodeIds.add(nodeId);
+        const adjacencNodes = this.getAdjacentNodes(nodeId);
```
Comment (foot) [inline client/app/scripts/stores/app-store.js]: `adjacentNodes`

**Comment by foot** [comment]
Edges not going both ways but not sure if this is a regression

botways

**Comment by foot** [comment]
but other than that LGTM.

**Comment by davkal** [comment]
> Edges not going both ways but not sure if this is a regression

Not a regression. In this scenario `192...` has `scope-app` in its adjacency list, but not the other way around. To fix this for a given `nodeA` , we need to check the adjacency list of all nodes, to see if `nodeA` is in any of their adjacency lists. This is already done a few lines above, so I can reuse that result.

**Comment by foot** [comment]
noice, lgtm

**Pull request merged**